salt/metalk8s/kubernetes/apiserver/installed.sls

```diff
@@ -26,6 +26,11 @@ include:
 {%- endif %}
 {%- set etcd_servers = etcd_servers | unique %}
 
+{%- set feature_gates = [] %}
+{%- for feature, value in pillar.kubeApiServer.get("featureGates", {}).items() %}
+{%-   do feature_gates.append(feature ~ "=" ~ value) %}
```

Comment:
This code now expects the values to be strings (which they should be). However, one could easily put something like
```
kubernetes:
  apiServer:
    featureGates:
      ephemeralContainers: true
```
in the configuration file, which turns the value into a bool instead of a string, which would be stringified in Python to `True` (I believe?), and `kube-apiserver` may not like that as an argument to `--feature-gates=...`.

So, we should either cater for such user 'mistake', or bail out.

Reply:
Yes but it works even with bool it's ok
Jinja will convert it to string

Reply:
Anyway ok I will convert bool to "true" or "false"